contracts/eth/nearbridge/contracts/AdminControlled.sol

```diff
@@ -1,30 +1,44 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
-pragma solidity ^0.8;
+pragma solidity 0.8.7;
 
-contract AdminControlled {
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+
+contract AdminControlled is AccessControlUpgradeable {
     address public admin;
     uint public paused;
 
-    constructor(address _admin, uint flags) {
-        admin = _admin;
-        paused = flags;
-    }
+    bytes32 public constant PAUSE_ROLE = keccak256("NEAR_BRIDGE_PAUSE");
```

Comment:
@idirall22 The AdminControlled is a top-level generic contract that can be used by any contract, so I think it is better to use another generic key name for PAUSE_ROLE, eg ("PAUSE_ROLE")